### PR TITLE
DOP-1857: Allow "reusable" references

### PR DESCRIPTION
**Pull request by sophstad**
[DOP-1857] Allow references/links defined with only a single underscore to be reused by name elsewhere in the corpus.

### CODE REVIEW

File: snooty/n.py [hunk 1]
```diff
@@ -368,6 +368,14 @@ class Reference(InlineParent):
     refname: str
 
 
+@dataclass
+class NamedReference(InlineParent):
```
Comment (i80and) [inline snooty/n.py:372]: What children could this have?

File: snooty/n.py [hunk 2]
```diff
@@ -368,6 +368,14 @@ class Reference(InlineParent):
...
+    __slots__ = ("refname", "refuri")
+    type = "named_reference"
+    refname: str
+    refuri: str
```
Comment (i80and) [inline snooty/n.py:376]: I... think we can actually remove the `refuri` field from `Target`, now

File: snooty/postprocess.py [hunk 3]
```diff
@@ -267,6 +267,36 @@ def get_include_argument(node: n.Directive) -> str:
         node.children = deep_copy_children
 
 
+class NamedReferenceHandler:
```
Comment (i80and) [inline snooty/postprocess.py:270]: Could we have a docstring briefly talking about what this does?

**Review by i80and — CHANGES_REQUESTED** [review]
A couple questions

Also, I think we need to add a method to `builders.man.TroffNodeHandler` to handle this node; e.g. https://github.com/mongodb/snooty-parser/blob/master/snooty/builders/man.py#L391-L392

File: snooty/n.py (hunk 1 above)
Comment (sophstad) [reply snooty/n.py:372]: Ah duh, copypasta mistake 😅

File: snooty/n.py [hunk 4]
```diff
@@ -343,7 +343,6 @@ class Target(Parent[Node]):
     type = "target"
     domain: str
     name: str
-    refuri: Optional[str]
```
Comment (i80and) [inline snooty/n.py:346]: We can also remove this from __slots__ :D

**Review by i80and — APPROVED** [review]

**Pull request merged**